File: java/Prototype/src/Car.java
public class Car implements PrototypeCloneable {
	private String name;
	private int engineCapacity;
	private int noOfSeats;
	private int horsePower;
	private boolean cruiseControl = false;
	private boolean automaticClimateControl = false;

	public Car() {
		CarDB carDB = new CarDB();
		this.engineCapacity = carDB.getEngineCapacity();
		this.noOfSeats = carDB.getNoOfSeats();
		this.horsePower = carDB.getNoOfSeats();
		System.out.println("Creating car object..");
	}

	public int getEngineCapacity() {
		return engineCapacity;
	}

	public int getNoOfSeats() {
		return noOfSeats;
	}

	public int getHorsePower() {
		return horsePower;
	}

	public boolean isCruiseControl() {
		return cruiseControl;
	}

	public void setCruiseControl(boolean cruiseControl) {
		this.cruiseControl = cruiseControl;
	}

	public boolean isAutomaticClimateControl() {
		return automaticClimateControl;
	}

	public void setAutomaticClimateControl(boolean automaticClimateControl) {
		this.automaticClimateControl = automaticClimateControl;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Car clone() throws CloneNotSupportedException {
		System.out.println("Cloning Car object..");
		return (Car) super.clone();
	}

	public void printDetails() {
		System.out.println("Name: " + this.name + "\nEngine Capacity: " + this.engineCapacity + "\nNumber of seats: "
				+ this.noOfSeats + "\nHorse Power: " + this.horsePower + "\nCruise Control: " + this.cruiseControl
				+ "\nAutomatic Climate Control: " + this.automaticClimateControl);
	}
}

File: ruby/Prototype/car.rb
class Car < PrototypeCloneable
  private
      attr_accessor :name
      attr_accessor :engineCapacity
      attr_accessor :noOfSeats
      attr_accessor :horsePower
      attr_accessor :cruiseControl
      attr_accessor :automaticClimateControl

  def initialize
    @cruiseControl = false
    @automaticClimateControl = false
    carDB = CarDB.new
    @engineCapacity = carDB.getEngineCapacity
    @noOfSeats = carDB.getNoOfSeats
    @horsePower = carDB.getNoOfSeats
    print "Creating car object..";
  end


  def clone
    print "Cloning Car object.."
    return (Car) super.clone();
  end

  def printDetails
    print "Name: #{@name} \nEngine Capacity: #{@engineCapacity} \nNumber of seats: #{@noOfSeats}
    \nHorse Power: #{@horsePower} \nCruise Control: #{@cruiseControl}
    \nAutomatic Climate Control: #{@automaticClimateControl}"
  end
end